sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -206,15 +210,114 @@ def convert_genbank_to_sbol3(
             doc.write(fpath=sbol3_file, file_format=sbol3.SORTED_NTRIPLES)
         return doc
 
+    def convert_sbol3_to_genbank(
+        self,
+        sbol3_file: str,
+        doc: sbol3.Document = None,
+        gb_file: str = "genbank.out",
+        write: bool = False,
+    ) -> List[SeqRecord]:
+        """Convert a SBOL3 document on disk into a GenBank document
+        The GenBank document is made using an array of SeqRecords using BioPython, by parsing SBOL3 objects
+
+        :param sbol3_file: path to read SBOL3 file from
+        :param gb_file: path to write GenBank file to, if write set to true
+        :param write: writes the generated genbank document to provided path
+        :return: Array of SeqRecord objects which comprise the generated GenBank document
+        """
+        if not doc:
+            doc = sbol3.Document()
+            doc.read(sbol3_file)
+        SEQ_RECORDS = []
+        logging.info(
+            "Creating GenBank and SO ontologies mappings for sequence feature roles"
+        )
+        # create updated py dict to store mappings between gb and so ontologies
+        map_created = self.create_GB_SO_role_mappings(
+            so2gb_csv=SO2GB_MAPPINGS_CSV, convert_gb2so=False
+        )
+        if not map_created:
+            # TODO: Need better SBOL3-GenBank specific error classes in future
+            raise ValueError(
+                f"Required CSV data files are not present in your package.\n    Please reinstall the sbol_utilities package.\n \
+                Stopping current conversion process.\n    Reverting to legacy converter if new Conversion process is not forced."
+            )
+        # consider sbol3 objects which are components
+        logging.info(f"Parsing SBOL3 Document components using SBOL3 Document: \n{doc}")
+        for obj in doc.objects:
+            if isinstance(obj, sbol3.Component):
+                SEQ_REC_FEATURES = []
+                logging.info(f"Parsing component - `{obj.display_id}` in sbol3 document.")
+                # TODO: can component have multiple sequences? How should we handle Seq objects for those
+                obj_seq = doc.find(obj.sequences[0])
+                seq = Seq(obj_seq.elements.upper())
+                # TODO: "Version" annotation information currently not stored when converted genbank to sbol3
+                seq_rec = SeqRecord(
+                    seq=seq,
+                    id=obj.display_id,
+                    description=obj.description,
+                    name=obj.display_id,
+                )
+                # TODO: hardcoded molecule_type as DNA, derivation?
+                seq_rec.annotations["molecule_type"] = "DNA"
+                # TODO: hardcoded topology as linear, derivation?
+                seq_rec.annotations["topology"] = "linear"
+                if obj.features:
+                    # converting all sequence features
+                    for obj_feat in obj.features:
```

Comment:
Add a filter for taking only the SequenceFeature features.
Add a TODO for SubComponent features in the future.